samples/nrf9160/lte_ble_gateway/src/main.c

```diff
@@ -427,34 +511,54 @@ static void modem_configure(void)
 static void sensors_init(void)
 {
 	int err;
-	const struct device *gps_dev = device_get_binding(CONFIG_GPS_DEV_NAME);
-	struct gps_config gps_cfg = {
-		.nav_mode = GPS_NAV_MODE_PERIODIC,
-		.interval = CONFIG_GPS_SEARCH_INTERVAL,
-		.timeout = CONFIG_GPS_SEARCH_TIMEOUT,
-	};
 
-	if (gps_dev == NULL) {
-		printk("Could not get %s device\n", CONFIG_GPS_DEV_NAME);
+	LOG_INF("Initializing GNSS");
+
+	nrf_modem_gnss_fix = false;
+
+	/* Configure GNSS. */
+	err = nrf_modem_gnss_event_handler_set(gnss_event_handler);
+	if (err) {
+		LOG_ERR("Failed to set GNSS event handler");
 		return;
 	}
 
-	err = gps_init(gps_dev, gps_handler);
+	/* Enable position NMEA GGA messages only. */
+	uint16_t nmea_mask = NRF_MODEM_GNSS_NMEA_GGA_MASK;
+
+	err = nrf_modem_gnss_nmea_mask_set(nmea_mask);
 	if (err) {
-		printk("Could not initialize GPS, error: %d\n", err);
+		LOG_ERR("Failed to set GNSS NMEA mask");
+		return;
+	}
+
+	/* This use case flag should always be set. */
+	uint8_t use_case = NRF_MODEM_GNSS_USE_CASE_MULTIPLE_HOT_START;
+
+	if (nrf_modem_gnss_use_case_set(use_case) != 0) {
+		LOG_WRN("Failed to set GNSS use case");
+	}
```

Comment:
nit: I would rather use this pattern. It's more common and I find it easier to read since you don't put code in an `if` statement test condition.
```
err = foo()
if (err) {
}
```